src/Microsoft.Maintainability.Analyzers/Core/AvoidUninstantiatedInternalClasses.cs

```diff
@@ -37,6 +43,205 @@ public override void Initialize(AnalysisContext analysisContext)
 
             // TODO: Configure generated code analysis.
             //analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
```

Comment:
You should uncomment this line and configure it with GeneratedCodeAnalysisFlags.Analyze - basically signup to receive callbacks for generated code (so we can note the type instantiations in generated code), but don't report diagnostics on types defined in generated code.